content/en/blog/2025/observing-lambdas/index.md

```diff
@@ -0,0 +1,122 @@
+services. With the
+[opentelemetry-lambda](https://github.com/open-telemetry/opentelemetry-lambda/blob/main/collector/README.md)
+extension layer, you get a local endpoint to send data to while it keeps track
+of the Lambda lifecycle and ensures your telemetry gets to the storage layer.
+
+## How does it work?
+
+When your function is called for the first time, the extension layer starts an
+instance of the OpenTelemetry Collector. The Collector build is a stripped down
+version, providing only components necessary in the context of Lambda. It
+registers with the Lambda
+[Extensions API](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html)
+and
+[Telemetry API](https://docs.aws.amazon.com/lambda/latest/dg/telemetry-api.html).
+By doing this, it receives notifications whenever your function is executed,
+emits a logline, or the execution context is about to be shut down.
+
+### This is where the magic happens
+
+Up until now, this just seems like extra work for nothing. You'll still have to
+wait for the Collector to ship the data, right? This is where the special
+`decouple` processor comes in. It separates the receiving and exporting
+components while interfacing with the Lambda lifecycle. This allows for the
+Lambda to return, even if not all data has been sent. At the next invocation (or
+on shutdown) the Collector continues shipping the data while your function does
+its thing.
+
+{{< figure src="diagram-execution-timing.svg" caption="Diagram showcasing how execution timing differs with and without a Collector">}}
+
+## How can I use it?
+
+As of November 2024, the opentelemetry-lambda project publishes
+[releases of the Collector extension layer](https://github.com/open-telemetry/opentelemetry-lambda/releases/tag/layer-collector%2F0.12.0).
+It can be configured through a configuration file hosted either in an S3 bucket
+or on an arbitrary HTTP server. It is also possible to bundle the configuration
+file with your Lambda code. In both cases, you have tradeoffs to consider.
+Remote configuration files add to the cold start duration as an additional
+request needs to be made, while bundling the configuration increases the
+management overhead when trying to control the configuration for multiple
+Lambdas.
+
+The simplest way to get started is with an embedded configuration. For this, add
+a file called `collector.yaml` to your function. This is a regular Collector
+configuration file. To take advantage of the Lambda specific extensions, they
+need to be configured. As an example, the following configuration receives
+traces and logs from the Telemetry API and sends them to another endpoint:
+
+```yaml
+# The `decouple` processor is configured by default if omitted.
+# It is explicitly added in this example to illustrate the entire pipeline
+# More information can be found at
+# https://github.com/open-telemetry/opentelemetry-lambda/tree/main/collector#auto-configuration
+
```

Comment:
Let's pull out all of that prose from the YAML file.

~~~suggestion
need to be configured. As an example, the configuration shown next receives
traces and logs from the Telemetry API and sends them to another endpoint.
The `decouple` processor is configured by default if omitted.
It is explicitly added in this example to illustrate the entire pipeline
For more information, see
[Auto-configuration](https://github.com/open-telemetry/opentelemetry-lambda/tree/main/collector#auto-configuration).

```yaml
~~~

If you prefer, you can move the prose to be _after_ the YAML.